### PR TITLE
METRON-1662: Adding download button 

### PR DESCRIPTION
**Pull request by tiborm**
## Contributor Comments
PR contains the implementation of the PCAP download button.

screen shot 2018-07-19 at 6 29 51 pm

The download button refers to and makes a GET request to the page file which selected by the pagination.

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  

In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel).
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [x] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```

- [x] Have you written or updated unit tests and or integration tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

### CODE REVIEW

**Comment by cestella** [comment]
Due to a bug in apache rat, our license checking component, it's ignoring all license checks on typescript files.  I have submitted a PR to up the rat plugin version and have fixed it in master #1126, let's make sure to add license headers here for new typescript files added in this PR though.

**Comment by mmiklavc** [comment]
>  Have you included steps to reproduce the behavior or problem that is being changed or addressed?
> 
>  Have you included steps or a guide to how the change may be verified and tested manually?

Can you submit or link to the test script/instructions for this?

**Comment by tiborm** [comment]
@cestella license headers added.

**Comment by tiborm** [comment]
@mmiklavc I updated the JIRA ticket with instructions for testing.
https://issues.apache.org/jira/browse/METRON-1662

File: metron-interface/metron-alerts/src/app/pcap/pcap-panel/pcap-panel.component.ts [hunk 1]
```diff
@@ -22,33 +22,35 @@ import { PcapRequest } from '../model/pcap.request'
...
-  ngOnInit() {
-  }
+  constructor(private pcapService: PcapService ) {}
 
   onSearch(pcapRequest) {
     console.log(pcapRequest);
     this.pdml = null;
     this.progressWidth = 0;
     this.pcapService.submitRequest(pcapRequest).subscribe(id => {
+      this.queryId = id;
```
Comment (merrimanr) [inline metron-interface/metron-alerts/src/app/pcap/pcap-panel/pcap-panel.component.ts]: The submitRequest function should return an observable of type PcapStatusResponse.

File: metron-interface/metron-alerts/src/app/pcap/pcap-panel/pcap-panel.component.ts [hunk 2]
```diff
@@ -22,33 +22,35 @@ import { PcapRequest } from '../model/pcap.request'
 import { Pdml } from '../model/pdml'
 import {Subscription} from "rxjs/Rx";
 
+class Query {
```
Comment (merrimanr) [inline metron-interface/metron-alerts/src/app/pcap/pcap-panel/pcap-panel.component.ts]: What is the purpose of this?

File: metron-interface/metron-alerts/src/app/pcap/pcap-panel/pcap-panel.component.ts (hunk 1 above)
Comment (tiborm) [reply metron-interface/metron-alerts/src/app/pcap/pcap-panel/pcap-panel.component.ts]: Nice catch, thanks! Turned this into PcapStatusResponse.

File: metron-interface/metron-alerts/src/app/pcap/pcap-panel/pcap-panel.component.ts (hunk 2 above)
Comment (tiborm) [reply metron-interface/metron-alerts/src/app/pcap/pcap-panel/pcap-panel.component.ts]: Thanks Ryan! It's dead code left there from some previous state of impl. Removed.

**Comment by merrimanr** [comment]
Thanks @tiborm looks good.  +1

**Comment by tiborm** [comment]
Thanks Guys!

**Pull request closed without merging**